pyface/ui/wx/color_dialog.py

```diff
@@ -50,10 +50,9 @@ def _create_contents(self, parent):
     # ------------------------------------------------------------------------
 
     def close(self):
-        if self.control.GetReturnCode() == wx.ID_OK:
```

Comment:
I don't understand why removing this check gives the expected behavior on Mac OS. This feels counter-intuitive.

Reply:
It turns out it was either not returning, or had not set, the return code at the point where that's being run.  I was seeing the same thing with the Font dialog (I'd click "OK" but the value returned was not "ID_OK").  Don't know if it's a Wx bug or what, but there's no real problem with always setting the color value.